From 0b4c4e080fa449946e56d9d6459e40861b746445 Mon Sep 17 00:00:00 2001
From: Serguei Katkov <serguei.i.katkov@intel.com>
Date: Fri, 2 Aug 2013 15:59:00 +0700
Subject: Dalvik: Fix the computation of gDvmJit.codeCacheByteUsed in PCG backend

BZ: 126551

PCG backend adds extra 1 byte for codeCacheByteUsed for each compiled trace.
It can result in bigger value for codeCacheByteUsed than code cache size.
That can lead to crash during cleaning the code cache.

Category: device-enablement
Domain: AOSP-Dalvik-Compiler-CG
Origin: internal
Upstream-Candidate: no, needs rework

Change-Id: Ia3e8f8586e69183119bbe5d38ebc71bc58392d26
Orig-MCG-Change-Id: I64553c5ab08476c89ab0d62195b96d161f9d2746
Signed-off-by: Serguei Katkov <serguei.i.katkov@intel.com>
Signed-off-by: Qiming Shi <qiming.shi@intel.com>
Reviewed-on: http://android.intel.com:8080/124379
Reviewed-by: Chen, Dong-Yuan <dong-yuan.chen@intel.com>
Tested-by: Chen, Dong-Yuan <dong-yuan.chen@intel.com>
Reviewed-by: cactus <cactus@intel.com>
Tested-by: cactus <cactus@intel.com>
Signed-off-by: Serguei Katkov <serguei.i.katkov@intel.com>
---
 vm/compiler/codegen/x86/pcg/CodeGeneration.cpp |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/vm/compiler/codegen/x86/pcg/CodeGeneration.cpp b/vm/compiler/codegen/x86/pcg/CodeGeneration.cpp
index c7c6fbd..e5ab789 100644
--- a/vm/compiler/codegen/x86/pcg/CodeGeneration.cpp
+++ b/vm/compiler/codegen/x86/pcg/CodeGeneration.cpp
@@ -1283,7 +1283,7 @@ void dvmCompilerPcgEmitCode (CompilationUnitPCG *cUnit, JitTranslationInfo* info
     // Increment gDvmJit.numCompilations to indicate successful compilation.
     cUnit->baseAddr = startAddr;
     info->codeAddress = startAddr;
-    cUnit->totalSize = currCachePtr - cacheStartPtr + 1;
+    cUnit->totalSize = currCachePtr - cacheStartPtr;
     gDvmJit.codeCacheByteUsed += cUnit->totalSize;
     gDvmJit.numCompilations++;
 
-- 
1.7.4.1

